magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/LSTnsb.py

```diff
@@ -0,0 +1,180 @@
+"""
+Evaluates NSB level for a LST run
+"""
+import argparse
+import glob
+import logging
+import os
+
+import numpy as np
+import yaml
+from lstchain.image.modifier import calculate_noise_parameters
+
+__all__ = ["nsb"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def nsb(run_list, simtel, lst_config, run_number):
+
+    """
+    Here we compute the NSB value for a run based on a subset of subruns.
+
+    Parameters
+    ----------
+    run_list : list
+        List of subruns in the run
+    simtel : str
+        Simtel (MC) file to be used to evaluate the extra noise in dim pixels
+    lst_config : str
+        LST configuration file (cf. lstchain)
+    run_number : int
+        LST run number
+
+    Returns
+    -------
+    list
+        List of the sub-run wise NSB values
+    """
+
+    noise = []
+    denominator = 25
+    if len(run_list) == 0:
+        logger.warning(
+            "There is no subrun matching the provided run number. Check the list of the LST runs (LST_runs.txt)"
+        )
+        return
+    if len(run_list) < denominator:
+        mod = 1
+    else:
+        mod = int(len(run_list) / denominator)
+    failed = 0
+    for ii in range(0, len(run_list)):
+        subrun = run_list[ii].split(".")[-2]
+        if mod == 0:
+            break
+        if ii % mod == 0:
+            try:
+                a, _, _ = calculate_noise_parameters(simtel, run_list[ii], lst_config)
+                noise.append(a)
+                logger.info(a)
+            except IndexError:
+                failed = failed + 1
+                if len(run_list) > denominator:
+                    mod = int(len(run_list) / (denominator + failed))

```

Comment:
since you modify 'mod' in the middle of processing over the files you will probably not end up with exactly denominator entries, 
Maybe something like ```mod = (len(run_list) - ii) // (denominator -len(noise))``` would work better, but either way it is not a big problem.

Reply:
Changed